src/python/WMCore/Services/WorkQueue/WorkQueue.py

```diff
@@ -106,12 +111,27 @@ def getChildQueuesAndPriority(self, stale=True):
             options['stale'] = 'update_after'
 
         data = self.db.loadView('WorkQueue', 'jobsByChildQueueAndPriority', options)
-        result = {}
+        result = []
         for x in data.get('rows', []):
-            result[x['key'][0]] = {x['key'][1]: x['value']}
+            item = {'agent_name': self._getShortName(x['key'][0]),
+                    'priority': int(x['key'][1])}
+            item.update(x['value'])
+            result.append(item)
 
         return result
 
+    def _getShortName(self, longQueueName):
+        """
+        Get a full workqueue queue name (full hostname + port) and return its short name,
+        otherwise it fails to get injected into elastic search. E.g.:
+            from "http://cmssrv217.fnal.gov:5984" to "cmssrv217"
+        """
+        if longQueueName is None:
+            return "AgentNotDefined"
```

Comment:
This means element is not pulled to agent and still "Available" state in global lqueue, right?